### PR TITLE
Added test to ensure extension rules pin the right rules_rust version

### PR DESCRIPTION
**Pull request by UebelAndre**
A test will now fail if the version is not updated in the extension rules.
```
==================== Test output for //test/bzlmod:bzl_ext_version_test:

running 1 test
test module_bzl_has_correct_version ... FAILED

failures:

---- module_bzl_has_correct_version stdout ----
    name = "rules_rust",
thread 'module_bzl_has_correct_version' panicked at external/rules_rust~/test/bzl_version/bzl_ext_version_test.rs:76:5:
assertion `left == right` failed: Core rules_rust and the dependency for the current module are out of sync.
  left: "0.55.0"
 right: "0.54.1"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    module_bzl_has_correct_version

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

### CODE REVIEW

File: test/bzl_version/bzl_ext_version_test.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    let core_version = {
+        let path = runfiles::rlocation!(r, std::env::var("CORE_MODULE_BAZEL").unwrap()).unwrap();
+        let text = std::fs::read_to_string(path).unwrap();
+        parse_module_bazel_version(&text)
+    };
+    let requested_version = {
+        let path = runfiles::rlocation!(r, std::env::var("MODULE_BAZEL").unwrap()).unwrap();
+        let text = std::fs::read_to_string(path).unwrap();
+        parse_rules_rust_version(&text)
```
Comment (illicitonion) [inline test/bzl_version/bzl_ext_version_test.rs:72]: Shouldn't we also check that the module() call _also_ has the same version? I thought our plan was to always release these together?

Comment (UebelAndre) [reply test/bzl_version/bzl_ext_version_test.rs:72]: Ooops, yup! Added the extra test!

Comment (illicitonion) [reply test/bzl_version/bzl_ext_version_test.rs:75]: Weird, I don't think I see it?

File: extensions/bindgen/test/bzlmod/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,12 @@
+load("@rules_rust//test/bzl_version:bzl_ext_version_test.bzl", "bzl_ext_version_test")
+load("@rules_rust//test/bzl_version:bzl_version_test.bzl", "bzl_version_test")
+
+bzl_ext_version_test(
+    name = "bzl_ext_version_test",
+    module_bazel = "//:MODULE.bazel",
+)
+
+bzl_version_test(
```
Comment (UebelAndre) [inline extensions/bindgen/test/bzlmod/BUILD.bazel:9]: These tests check the version in `MODULE.bazel` is the same as rules_rust.

File: test/bzl_version/bzl_ext_version_test.rs (hunk 1 above)
Comment (UebelAndre) [reply test/bzl_version/bzl_ext_version_test.rs:75]: https://github.com/bazelbuild/rules_rust/pull/3052/files#r1872478871

Comment (UebelAndre) [reply test/bzl_version/bzl_ext_version_test.rs:75]: For example, if I apply the following diff
```diff
diff --git a/extensions/prost/MODULE.bazel b/extensions/prost/MODULE.bazel
index 81a5121c..dc926be7 100644
--- a/extensions/prost/MODULE.bazel
+++ b/extensions/prost/MODULE.bazel
@@ -2,7 +2,7 @@

 module(
     name = "rules_rust_prost",
-    version = "0.55.0",
+    version = "0.54.1",
 )

 bazel_dep(
```
This test fails
```
INFO: From Testing //private/tests/bzlmod:bzl_version_test:
==================== Test output for //private/tests/bzlmod:bzl_version_test:

running 1 test
test module_bzl_has_correct_version ... FAILED

failures:

---- module_bzl_has_correct_version stdout ----
thread 'module_bzl_has_correct_version' panicked at external/rules_rust~/test/bzl_version/bzl_version_test.rs:36:5:
assertion `left == right` failed: //:version.bzl and //:MODULE.bazel versions are out of sync
  left: "0.55.0"
 right: "0.54.1"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    module_bzl_has_correct_version

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
```

**Review by illicitonion — APPROVED** [review]
Thanks!

**Pull request merged**